public/scripts/utils.js

```diff
@@ -44,4 +44,99 @@ module.exports = {
       .normalize('NFD')
       .replace(/[\u0300-\u036f]/g, '');
   },
+
+  sortByColors: function (colors) {
+    const isGray = (rgb, range) => {
+      var { r, g, b } = { ...rgb };
+      return (
+        r >= g - range &&
+        r <= g + range &&
+        b >= g - range &&
+        b <= g + range &&
+        r >= b - range &&
+        r <= b + range
+      );
+    };
+
+    const rgbToHsl = (rgb) => {
+      var { r, g, b } = { ...rgb };
+      // Make r, g, and b fractions of 1
+      r /= 255;
+      g /= 255;
+      b /= 255;
+
+      // Find greatest and smallest channel values
+      let cmin = Math.min(r, g, b),
+        cmax = Math.max(r, g, b),
+        delta = cmax - cmin,
+        h = 0,
+        s = 0,
+        l = 0;
+
+      // Calculate hue
+      // No difference
+      if (delta == 0) h = 0;
+      // Red is max
+      else if (cmax == r) h = ((g - b) / delta) % 6;
+      // Green is max
+      else if (cmax == g) h = (b - r) / delta + 2;
+      // Blue is max
+      else h = (r - g) / delta + 4;
+
+      h = Math.round(h * 60);
+
+      // Make negative hues positive behind 360°
```

Comment:
I don't get the "behind 360" part of this comment. Negative hues will never be greater than 360 after the following line. Maybe just say

```suggestion
      // Ensure all hues are positive
```